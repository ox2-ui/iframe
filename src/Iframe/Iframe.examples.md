### Iframe

    <Iframe url={'http://apple.com'} />
