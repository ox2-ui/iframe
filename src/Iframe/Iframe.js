import PropTypes from 'prop-types';
import React from 'react';
import { AutoSizer } from 'react-virtualized';

const styles = {
  frame: {
    border: 'none',
    margin: '0',
    padding: '0',
  },
};

/**
 * Iframe Component
 */
const Iframe = props => {
  const { className, style, url } = props;

  const mergedStyles = {
    ...styles.frame,
    ...style,
  };

  return (
    <AutoSizer>
      {({ height, width }) =>
        <iframe
          className={className}
          height={height}
          src={url}
          style={mergedStyles}
          width={width}
        >
          <p>Your browser does not support iframes.</p>
        </iframe>}
    </AutoSizer>
  );
};

Iframe.propTypes = {
  /**
   * The css class name of the iframe element.
   */
  className: PropTypes.string,
  /**
   * The css style object of the iframe element.
   */
  style: PropTypes.object,
  /**
   * Url to load into the iframe
   */
  url: PropTypes.string,
};

export default Iframe;
