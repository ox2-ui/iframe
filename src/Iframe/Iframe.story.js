import React from 'react';
import { storiesOf } from '@storybook/react';
import Iframe from './Iframe';

const styles = {
  wrapper: {
    width: '1200px',
    height: '800px',
  },
};

storiesOf('Iframe', module).add('default', () =>
  <div style={styles.wrapper}>
    <Iframe url={'sample/index.html'} />
  </div>,
);
