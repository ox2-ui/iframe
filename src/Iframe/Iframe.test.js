/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Iframe from './Iframe';

it('renders correctly', () => {
  const wrapper = mount(
    <Iframe url={'sample/index.html'} />
  );

  expect(wrapper).toMatchSnapshot();
});
