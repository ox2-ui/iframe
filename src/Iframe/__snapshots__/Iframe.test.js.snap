exports[`test renders correctly 1`] = `
<Iframe
  url="sample/index.html">
  <AutoSizer
    onResize={[Function]}>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }>
      <iframe
        height={0}
        src="sample/index.html"
        style={
          Object {
            "border": "none",
            "margin": "0",
            "padding": "0",
          }
        }
        width={0}>
        <p>
          Your browser does not support iframes.
        </p>
      </iframe>
    </div>
  </AutoSizer>
</Iframe>
`;
